application/auth/controller.py

```diff
@@ -0,0 +1,98 @@
+from flask import jsonify, make_response, request
+from flask_httpauth import HTTPTokenAuth
+from flask_restful import Resource, Api
+
+from application import db
+from application.auth.models import User
+
+
+class Register(Resource):
+    def post(self):
+        try:
+            username = request.json["username"]
+            password = request.json["password"]
+            if not username:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Username is missing."
+
+                        }
+                    ), 400)
+            if not password:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Password is missing."
+
+                        }
+                    ), 400)
+
+            user = User.query.filter_by(username=username).first()
+            if user:
+                return make_response(
+                    jsonify({
+                        'message': "User with the username already exists.",
+                        "username": username
+
+                    }), 409)  # existing user
+            user = User(username, password)
+            db.session.add(user)
+            db.session.commit()
+            return make_response(
+                jsonify({
+                    'message': "User registration successful.",
+                    "username": username,
+
+
+                }), 200)
+        except:
+            return make_response(jsonify({
+                "error": "missing data in request."
+            }))
```

Comment:
consider adding status code i.e `400` to denote the error that happened in the request